src/main/java/raisetech/StudentManagement/service/StudentService.java

```diff
@@ -111,12 +112,24 @@ public void addStudent(StudentDetail studentDetail) {
   //生徒情報の変更
   @Transactional
   public void updateStudent(StudentDetail studentDetail) {
+    //引数からを生徒情報をオブジェクトにセット
     Student student = studentDetail.getStudent();
+    //生徒情報の更新処理
     repository.updateStudent(student);
+    //引数から受講コース情報をオブジェクトにセット
     List<CourseDetail> courseDetails = studentDetail.getCourseDetail();
+    //受講コース情報の更新処理
     for (CourseDetail courseDetail : courseDetails) {
       StudentsCourses studentsCourses = courseDetail.getStudentsCourses();
       repository.updateStudentCourses(studentsCourses);
     }
   }
+
+  @Transactional
+  public void deleteStudent(Student student) {
+    //削除フラグを立てる
+    student.setDeleted(true);
```

Comment:
個人的には
```suggestion
    student.delete();
```
にしてdeleteメソッド内で削除フラグtrueにしてもいいかと。
そのほうがより自然に読めますし、setterを消せば
```java
student.setDeleted(false);
```
という誤ったコードの記述を防げますしね。

Reply:
そうですね。
そもそもsetterは更新のためのものですから登録に使う必要はないですね。
なお、更新時であってもsetterを使うのは私は保守性の観点からあまり良くないと思ってます。

setter 不要

などで調べてみるとよいです😎

Reply:
修正いたしました。
930f701ea3abcba27374508a8e0dad14ad9c6609